kore/src/Kore/Builtin/Bool.hs

```diff
@@ -205,11 +206,47 @@ termAndEquals unifyChildren a b =
       , value1
       , Just BoolAnd { operand1, operand2 } <- matchBoolAnd termLike2
       , isFunctionPattern termLike2
-      = lift $ do
-        unification1 <- unifyChildren' termLike1 operand1
-        unification2 <- unifyChildren' termLike1 operand2
-        let conditions = unification1 <> unification2
-        pure (Pattern.withCondition termLike1 conditions)
+      = unifyBothWith unifyChildren' termLike1 operand1 operand2
+    worker _ _ = empty
+
+unifyBothWith
+    :: forall variable unifier
+    .  InternalVariable variable
+    => MonadUnify unifier
+    =>  (TermLike variable
+        -> TermLike variable
+        -> unifier (Pattern.Condition variable)
+        )
```

Comment:
`termAndEquals` and `termOrEquals` have the same definition of `unifyChildren'`, so let's move that into this function, which can then take a `TermSimplifier` as its argument:

```suggestion
    => TermSimplifier variable unifier
```